### PR TITLE
fix for megatron + oss

### PR DESCRIPTION
**Pull request by ngoyal2707**
# Before submitting

- [ ] Was this discussed/approved via a Github issue? (no need for typos, doc improvements)
- [ ] Did you read the [contributor guideline](https://github.com/facebookresearch/fairscale/blob/master/CONTRIBUTING.md)?
- [ ] Did you make sure to update the docs?
- [ ] Did you write any new necessary tests?

## What does this PR do?
Fixes # (issue).

## PR review
Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Review by blefaudeux — APPROVED** [review]
lgtm, sorry for the oversight

**Review by joshim5 — APPROVED** [review]
Naman and I discussed on chat, lgtm. Naman found that this fixes the inconsistencies w/ and w/o OSS. Tagging @blefaudeux since this fixes #122

File: fairscale/optim/oss.py [hunk 1]
```diff
@@ -434,14 +434,14 @@ def _broadcast_params(
             # Since all the parameters are already sorted per increasing size, we only need to consider the first ones.
             while i_bucketed < len(params) and offset + params[i_bucketed].numel() < buffer_size:
                 end = offset + params[i_bucketed].numel()
-                if rank == self_rank:
+                if global_rank == self_rank:
```
Comment (msbaines) [inline fairscale/optim/oss.py:437]: This function might be simpler if it didn't pass group and self_rank and instead just used self. That way, the correctness would be obvious. self_rank is ambiguous as to whether it is global or local.

```suggestion
                if global_rank == self.global_rank:
```

**Review by msbaines — APPROVED** [review]
Thanks for fixing. Some minor nits.

It would be nice to add a test so that this doesn't break in the future.

**Comment by blefaudeux** [comment]
(context) we discussed the regression testing with Joshua already, I've a task up for that T77211997

**Comment by blefaudeux** [comment]
Adding a test today, @joshim5 @ngoyal2707 would you prefer to merge as-is or pull in the suggestion from @msbaines ? I can also bring that in if you prefer

**Comment by ngoyal2707** [comment]
> Adding a test today, @joshim5 @ngoyal2707 would you prefer to merge as-is or pull in the suggestion from @msbaines ? I can also bring that in if you prefer

lets merge as-is? and we can follow up with tests and other changes? cause current implementation breaks silently.

**Pull request merged**